### ISSUE TITLE
Crashing while setting force decode policy.

### ISSUE THREAD

**Issue by NigamAnkit** [SDWebImage/SDWebImage#3702]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           |  ios
 Platform Version        | 17.0
 SDWebImage Version      | 5.19.1
 Integration Method      |  cocoapods 
 Xcode Version           | Xcode 15
 Repro rate              | all the time (100%) 

### Issue Description and Steps

I was using SDWebImage version 3.x as it was very stable and required no update from our App's perspective. But due to Apple Privacy Manifests required we updated to the latest 5.19.1 version.

Doing so we have to make many updates. 
OLD:

`SDImageCache.shared().maxCacheAge = 60 * 60 * 24`
`SDWebImageManager.shared()?.delegate = withDelegate`
`SDImageCache.shared().shouldDecompressImages = false`
`SDWebImageDownloader.shared().shouldDecompressImages = false`

New:
`SDImageCache.shared.config.maxDiskAge = 60 * 60 * 24`
        `SDWebImageManager.shared.delegate = withDelegate`
`SDImageCoderHelper.defaultDecodeSolution = .uiKit`
```
SDWebImageManager.shared.optionsProcessor = SDWebImageOptionsProcessor(block: { url, option, context in
            var updatedContext = context
            updatedContext?[.imageForceDecodePolicy] = [SDImageForceDecodePolicy.never]
            if let imageURL = url {
                if !imageURL.absoluteString.contains(appConfig.assetBaseURL.absoluteString) && !imageURL.absoluteString.contains(appConfig.assetBaseURLDev.absoluteString) {
                    updatedContext?[.imageTransformer] = HarkImageTransformer(size: CGSize(width: 150, height: 150), scaleMode: .aspectFill)
                }
            }
   return SDWebImageOptionsResult(options: option, context: updatedContext)
        })
```

also I implemented Class for ImageTransformer by subclassing.

```
class HarkImageTransformer: SDImageResizingTransformer {
    
    override func transformedImage(with image: UIImage, forKey key: String) -> UIImage? {
        if image.size.width > 4000 {
            
            return UIImage(imageLiteralResourceName: "no_album_art")
        }else {
            return super.transformedImage(with: image, forKey: key)
        }
    }
}
```

I tried to find where it is crashing and found that it crashed 

Also, Please update the documentation. Its so difficult to find right approach to use global force decode policy.

Detected the current OS's ImageIO PNG Decoder is buggy on indexed color PNG. Perform workaround solution...
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[Swift.__SwiftDeferredNSArray unsignedIntegerValue]: unrecognized selector sent to instance 0x6000003ad7e0'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001804658a8 __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x000000018005c09c objc_exception_throw + 56
	2   CoreFoundation                      0x000000018047a6f8 +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
	3   CoreFoundation                      0x00000001804699f8 ___forwarding___ + 1280
	4   CoreFoundation                      0x000000018046bd1c _CF_forwarding_prep_0 + 92
	5   SDWebImage                          0x0000000105d7ae3c SDImageLoaderDecodeImageData + 1284
	6   SDWebImage                          0x0000000105d89160 __95-[SDWebImageDownloaderOperation startCoderOperationWithImageData:pendingTokens:finishedTokens:]_block_invoke + 872
	7   Foundation                          0x0000000180d7b2b0 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
	8   Foundation                          0x0000000180d7b1a4 -[NSBlockOperation main] + 100
	9   Foundation                          0x0000000180d7de98 __NSOPERATION_IS_INVOKING_MAIN__ + 12
	10  Foundation                          0x0000000180d7a404 -[NSOperation start] + 620
	11  Foundation                          0x0000000180d7e604 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 12
	12  Foundation                          0x0000000180d7e314 __NSOQSchedule_f + 168
	13  libdispatch.dylib                   0x000000010791a038 _dispatch_block_async_invoke2 + 104
	14  libdispatch.dylib                   0x000000010790993c _dispatch_client_callout + 16
	15  libdispatch.dylib                   0x000000010790cddc _dispatch_continuation_pop + 756
	16  libdispatch.dylib                   0x000000010790bd68 _dispatch_async_redirect_invoke + 900
	17  libdispatch.dylib                   0x000000010791d4bc _dispatch_root_queue_drain + 372
	18  libdispatch.dylib                   0x000000010791e068 _dispatch_worker_thread2 + 256
	19  libsystem_pthread.dylib             0x0000000105e4383c _pthread_wqthread + 224
	20  libsystem_pthread.dylib             0x0000000105e4263c start_wqthread + 8
)
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[Swift.__SwiftDeferredNSArray unsignedIntegerValue]: unrecognized selector sent to instance 0x6000003ad2a0'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001804658a8 __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x000000018005c09c objc_exception_throw + 56
	2   CoreFoundation                      0x000000018047a6f8 +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
	3   CoreFoundation                      0x00000001804699f8 ___forwarding___ + 1280
	4   CoreFoundation                      0x000000018046bd1c _CF_forwarding_prep_0 + 92
	5   SDWebImage                          0x0000000105d7ae3c SDImageLoaderDecodeImageData + 1284
	6   SDWebImage                          0x0000000105d89160 __95-[SDWebImageDownloaderOperation startCoderOperationWithImageData:pendingTokens:finishedTokens:]_block_invoke + 872
	7   Foundation                          0x0000000180d7b2b0 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
	8   Foundation                          0x0000000180d7b1a4 -[NSBlockOperation main] + 100
	9   Foundation                          0x0000000180d7de98 __NSOPERATION_IS_INVOKING_MAIN__ + 12
	10  Foundation                          0x0000000180d7a404 -[NSOperation start] + 620
	11  Foundation                          0x0000000180d7e604 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 12
	12  Foundation                          0x0000000180d7e314 __NSOQSchedule_f + 168
	13  libdispatch.dylib                   0x000000010791a038 _dispatch_block_async_invoke2 + 104
	14  libdispatch.dylib                   0x000000010790993c _dispatch_client_callout + 16
	15  libdispatch.dylib                   0x000000010790cddc _dispatch_continuation_pop + 756
	16  libdispatch.dylib                   0x000000010790bd68 _dispatch_async_redirect_invoke + 900
	17  libdispatch.dylib                   0x000000010791d4bc _dispatch_root_queue_drain + 372
	18  libdispatch.dylib                   0x000000010791e068 _dispatch_worker_thread2 + 256
	19  libsystem_pthread.dylib             0x0000000105e4383c _pthread_wqthread + 224
	20  libsystem_pthread.dylib             0x0000000105e4263c start_wqthread + 8
)
libc++abi: terminating due to uncaught exception of type NSException
Message from debugger: killed

**Comment by dreampiggy** [comment]
`[Swift.__SwiftDeferredNSArray unsignedIntegerValue]`
Alreay point out the issue.

You don't pass `NSNumber *`. This API value is `NSNumber`, not `Swift.Int`

**Comment by dreampiggy** [comment]
```
updatedContext?[.imageForceDecodePolicy] = [SDImageForceDecodePolicy.never.rawValue]
```

or 

```
updatedContext?[.imageForceDecodePolicy] = NSNumber(integer: SDImageForceDecodePolicy.never.rawValue)
```

**Comment by dreampiggy** [comment]
Where is typo code come from ? `updatedContext?[.imageForceDecodePolicy] = [SDImageForceDecodePolicy.never]` ?
Is that written by me in somewhere ?

**Comment by NigamAnkit** [comment]
Thanks, I figured that out today and was about to close the issue by myself. But thanks for your reply.
Another suggestion is to please improve the documentation about using advance things like option processors like it was after so much search through the issues I found the Objective C code and converting to swift was not that straight as it seems.

And to you question "Where this typo code come from ? updatedContext?[.imageForceDecodePolicy] = [SDImageForceDecodePolicy.never] ?
Is that written by me in somewhere ?"

Yes, I got it from your Comment in the lib.
@deprecated Deprecated in v5.17.0, if you don't want force-decode, pass [.imageForceDecodePolicy] = [SDImageForceDecodePolicy.never] in context option.
<img width="1133" alt="Screenshot 2024-04-26 at 2 06 43 PM" src="https://github.com/SDWebImage/SDWebImage/assets/32325505/89f85390-431d-4011-bd4e-d9221f6374f4">

I don't know was it my bad to understand it otherwise.

Thank you for this beautiful Lib.
Can you also evaluate if all the code I wrote is the write way of using advance settings.

**Comment by dreampiggy** [comment]
Seems typo in documentation...(Sometimes it's hard to switch between Objcective-C / Swift syntax in the mind)

**Comment by dreampiggy** [comment]
Fixed by ea2e3d5debbe19f5f59284cf2c605c67dac3cdbd
